cmake/toolchain-gcc.cmake

```diff
@@ -67,16 +88,8 @@ set(CMAKE_C_FLAGS_DEBUG ${COMPILER_FLAGS_DEBUG})
 
 set(CMAKE_EXE_LINKER_FLAGS "")
 
-if (GCC_ENABLE_LEAK_CHECK)
-	check_linker_flag("-fsanitize=leak" SUPPORTS_FSANITIZE_LEAK)
-
-	if (SUPPORTS_FSANITIZE_LEAK)
-		set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fsanitize=leak")
-	endif()
-endif()
-
 if (SUPPORTS_SANITIZE_ADDRESS)
```

Comment:
Keeping this condition means `GCC_ENABLE_LEAK_CHECK` will fail to do anything on its own.